frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/capacity-breakdown/capacity-breakdown-card.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Dropdown, humanizeBinaryBytesWithoutB } from '@console/internal/components/utils';
+import {
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import { breakdownQueryMap, CAPACITY_BREAKDOWN_QUERIES } from '../../../../constants/queries';
+import { PROJECTS } from '../../../../constants/index';
+import { BreakdownCardBody } from '../breakdown-card/breakdown-body';
+import { HeaderPrometheusLink } from '../breakdown-card/breakdown-header';
+import { getStackChartStats } from '../breakdown-card/utils';
+import './capacity-breakdown-card.scss';
+
+const keys = Object.keys(breakdownQueryMap);
+const dropdownOptions = _.zipObject(keys, keys);
+
+const BreakdownCard: React.FC<DashboardItemProps> = ({
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+}) => {
+  const [metricType, setMetricType] = React.useState(PROJECTS);
+  const { queries, model, metric } = breakdownQueryMap[metricType];
+  const queryKeys = Object.keys(queries);
+
+  React.useEffect(() => {
+    queryKeys.forEach((q) => watchPrometheus(queries[q]));
+    return () => queryKeys.forEach((key) => stopWatchPrometheusQuery(queries[key]));
+  }, [watchPrometheus, stopWatchPrometheusQuery, metricType, queryKeys, queries]);
+
+  const results = queryKeys.map((key) => prometheusResults.getIn([queries[key], 'data']));
```

Comment:
It would be better if you divide the results into multiple variables so that it's easy to understand what the different queries are returning.

Reply:
Is it fine if we can take it up in a follow up PR ?

Reply:
sure